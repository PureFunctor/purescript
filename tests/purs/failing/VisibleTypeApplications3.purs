-- @shouldFailWith KindsDoNotUnify
module Main where

data Proxy :: forall k. k -> Type
data Proxy a = Proxy

fail :: forall @k (@t :: k). Proxy t
fail = Proxy

fail' = fail @Type @"NotType"
