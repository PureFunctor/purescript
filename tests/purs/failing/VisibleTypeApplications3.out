Error found:
in module [33mMain[0m
at tests/purs/failing/VisibleTypeApplications3.purs:10:9 - 10:30 (line 10, column 9 - line 10, column 30)

  Could not match kind
  [33m        [0m
  [33m  Symbol[0m
  [33m        [0m
  with kind
  [33m      [0m
  [33m  Type[0m
  [33m      [0m

while trying to match type [33m"NotType"[0m
  with type [33mt0[0m
while inferring the type of [33mfail[0m
in value declaration [33mfail'[0m

where [33mt0[0m is an unknown type

See https://github.com/purescript/documentation/blob/master/errors/KindsDoNotUnify.md for more information,
or to contribute content related to this error.

